### PR TITLE
[StorageQuota] Add more metrics and fix a statsheader copy bug.

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
This PR 
1. Add more metrics to `AccountStatsMySqlStore`
2. Fix a StatsHeader copy constructor bug
3. Add support for mysql aggregation task in HelixClusterWideAggregationTaskTool.

### CODE REVIEW

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixClusterWideAggregationTool.java [hunk 1]
```diff
@@ -167,8 +172,9 @@ public static void main(String[] args) throws Exception {
               String jobId = statsType.toString().toLowerCase() + (isRecurrentWorkflow ? RECURRENT_JOB_SUFFIX
                   : ONE_TIME_JOB_SUFFIX);
               String taskId = statsType.toString().toLowerCase() + TASK_SUFFIX;
-              String aggregationCommand =
-                  String.format("%s_%s", HelixHealthReportAggregatorTask.TASK_COMMAND_PREFIX, reportName);
+              String aggregationCommand = String.format("%s_%s",
+                  config.isMysql ? MySqlReportAggregatorTask.TASK_COMMAND_PREFIX
```
Comment (jsjtzyy) [inline ambry-tools/src/main/java/com/github/ambry/clustermap/HelixClusterWideAggregationTool.java:176]: Minor: I am not entirely sure what will happen if the report is not found as this report may be PartitionClass report which is temporarily not supported by mysql.

**Review by jsjtzyy — APPROVED** [review]
LGTM

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixClusterWideAggregationTool.java (hunk 1 above)
Comment (justinlin-linkedin) [reply ambry-tools/src/main/java/com/github/ambry/clustermap/HelixClusterWideAggregationTool.java:176]: I won't worry too much about it, it's a cli tool. If we make a mistake, we can always revert it.

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/MySqlReportAggregatorTask.java [hunk 2]
```diff
@@ -87,6 +87,7 @@ public TaskResult run() {
         statsWrappers.put(instanceName,
             accountStatsStore.queryStatsOf(clusterMapConfig.clusterMapClusterName, instanceName));
       }
+      logger.info(String.format("Aggregating stats from %d hosts", statsWrappers.size()));
```
Comment (zzmao) [inline ambry-clustermap/src/main/java/com/github/ambry/clustermap/MySqlReportAggregatorTask.java:90]: why not ```logger.info("Aggregating stats from {} hosts", statsWrappers.size());```

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/MySqlReportAggregatorTask.java [hunk 3]
```diff
@@ -99,6 +100,7 @@ public TaskResult run() {
         String recordedMonthValue = accountStatsStore.queryRecordedMonth(clusterMapConfig.clusterMapClusterName);
         if (recordedMonthValue == null || recordedMonthValue.isEmpty() || !currentMonthValue.equals(
             recordedMonthValue)) {
+          logger.info("Taking snapshot of aggregated stats for month " + currentMonthValue);
```
Comment (zzmao) [inline ambry-clustermap/src/main/java/com/github/ambry/clustermap/MySqlReportAggregatorTask.java:103]: same here, why not use `{}`

**Review by zzmao — COMMENTED** [review]
minor comments regarding logger.info()

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/MySqlReportAggregatorTask.java (hunk 2 above)
Comment (justinlin-linkedin) [reply ambry-clustermap/src/main/java/com/github/ambry/clustermap/MySqlReportAggregatorTask.java:90]: does it matter? is one better than the other?

Comment (lightningrob) [reply ambry-clustermap/src/main/java/com/github/ambry/clustermap/MySqlReportAggregatorTask.java:90]: String.format is generally redundant inside logger calls.

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixClusterWideAggregationTool.java [hunk 4]
```diff
@@ -108,6 +108,10 @@
     @Default("false")
     final boolean deleteSpecifiedWorkflow;
 
+    @Config("is.mysql")
+    @Default("false")
+    final boolean isMysql;
```
Comment (lightningrob) [inline ambry-tools/src/main/java/com/github/ambry/clustermap/HelixClusterWideAggregationTool.java:113]: Time to remove Helix from the class name?  (Unless that will break something.)

**Review by lightningrob — APPROVED** [review]

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixClusterWideAggregationTool.java (hunk 4 above)
Comment (justinlin-linkedin) [reply ambry-tools/src/main/java/com/github/ambry/clustermap/HelixClusterWideAggregationTool.java:113]: Helix in the name is fine. Although we start to read the container storage reports from mysql, but this tool will instruct helix to execute these tasks.

**Pull request merged**